src/globalstate.h

```diff
@@ -19,8 +19,32 @@
 
 #define MAXPRINTDEPTH 1024L
 
+#if defined(HPCGAP)
+#define TLS_NUM_EXTRA 256
+#endif
+
 typedef struct GlobalState
 {
+#if defined(HPCGAP)
+  int threadID;
+  void *threadLock;
+  void *threadSignal;
+  void *acquiredMonitor;
+  unsigned multiplexRandomSeed;
+  void *currentRegion;
+  void *threadRegion;
+  void *traversalState;
+  Obj threadObject;
+  Obj tlRecords;
+  Obj lockStack;
+  int lockStackPointer;
+  Obj copiedObjs;
+  Obj interruptHandlers;
+  void *CurrentHashLock;
+  char *CurrFuncName;
+  int DisableGuards;
+#endif
+
```

Comment:
Ah, I see: the other diffs are changed later on, in a different commit.

Reviewing this PR is very difficult: the full diff view is useless, as all the TLS->STATE renaming drowns out the other changes. And reviewing of  a per-commit basis is hard because the history is not cleaned up.

To solve this, I suggest splitting this PR: first have a PR which renames TLS to STATE and which does nothing else. Ideally, the renaming is done with a script which replaces `TLS(` by `STATE(` globally in `src` and `hpcgap/src`.

Then, a second PR based on the first contains the rest ofnthe chnages, which then can be reviewed based on the overall diff.